src/requestSchemas.ts

```diff
@@ -0,0 +1,157 @@
+  'check-system-version',
+  'force-system-version',
+  'manifest',
+  'include-components'
+];
+
+const UNSUPPORTED_DATA_REQ_ARGS = [
+  'expression',
+  'parameters',
+  'manifest',
+  'include-dependencies',
+  'include-components'
+];
+
+const UNSUPPORTED_SEARCH_ARGS = [
+  'date',
+  'effective',
+  'jurisdiction',
+  'context',
+  'context-type',
+  'context-type-quantity',
+  'context-type-value',
+  'topic',
+  'composed-of',
+  'depends-on',
+  'derived-from',
+  'successor',
+  'predecessor'
+];
+
+const customErrorMap: z.ZodErrorMap = (issue, ctx) => {
+  if (issue.code === z.ZodIssueCode.custom) {
+    if (issue.params?.type === 'NotImplemented') {
+      throw new NotImplementedError(issue.message ?? ctx.defaultError);
+    } else {
+      throw new BadRequestError(issue.message ?? ctx.defaultError, issue.params?.serverErrorCode);
+    }
+  } else if (issue.code === z.ZodIssueCode.unrecognized_keys) {
+    throw new BadRequestError(issue.message ?? ctx.defaultError, constants.ISSUE.CODE.VALUE);
+  }
+  throw new BadRequestError(issue.message ?? ctx.defaultError);
+};
+
+z.setErrorMap(customErrorMap);
+
+const hasIdentifyingInfo = (args: Record<string, any>) => args.id || args.url || args.identifier;
+
+/**
+ * Returns a function that checks if any unsupported params are present, then runs the
+ * other passed in functions in sequence. Each catchFunction is expected to check for
+ * invalid input and call ctx.addIssue if invalid input is detected. Any added issue will
+ * trigger an error to be thrown.
+ */
+export function catchInvalidParams(
+  catchFunctions: ((val: Record<string, any>, ctx: z.RefinementCtx) => void)[],
+  unsupportedArgs?: string[]
+) {
+  return (val: Record<string, any>, ctx: z.RefinementCtx) => {
+    const includedUnsupported = Object.keys(val).filter((v: string) => unsupportedArgs?.includes(v));
+    if (includedUnsupported.length > 0) {
+      ctx.addIssue({
+        code: z.ZodIssueCode.custom,
+        params: { type: 'NotImplemented' },
+        message: `Parameters: ${includedUnsupported.join(', ')} are not yet supported on this server`
+      });
+    } else {
+      catchFunctions.forEach(func => func(val, ctx));
+    }
+  };
+}
+
+/**
+ * Checks that either id, url, or identifier parameter is included. Adds an issue to the ctx
+ * that triggers a BadRequest to be thrown if not.
+ */
+export function catchMissingIdentifyingInfo(val: Record<string, any>, ctx: z.RefinementCtx) {
+  if (!hasIdentifyingInfo(val)) {
+    ctx.addIssue({
+      code: z.ZodIssueCode.custom,
+      params: { serverErrorCode: constants.ISSUE.CODE.REQUIRED },
+      message: 'Must provide identifying information via either id, url, or identifier parameters'
+    });
+  }
+}
+
+export const IdentifyingParameters = z
+  .object({
+    id: z.string(),
+    url: z.string().url(),
+    version: z.string(),
+    identifier: z.string()
+  })
+  .partial();
+
+export const PackageArgs = IdentifyingParameters.extend({
```

Comment:
Related: I actually realized a bug in the code unrelated to your PR. I would expect providing `include-terminology` as `'false'` in a request to not do the valueset resolution, but since everything comes through as a string over http, whenever the parameter is defined, the code doesn't care if its value is true or false. I think this can be solved in a couple ways, but maybe we can use a combo of [transform](https://zod.dev/?id=transform) and the actual output of the `parse` result.

For this specific case, something like 

```ts
  'include-terminology': z.enum(['true', 'false']).transform(val => val === 'true')
```

We could also try to to something similar for params that are integers, ensuring they are parsed as such. This will help us implement them later.

I also think that using the result of the parse is a good idea especially if we do this transform, as it will give us full type safety